internal/storage/image.go

```diff
@@ -398,6 +408,145 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+// nolint:staticcheck
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, userSpecifiedImage string) error {
+	userSpecifiedImageParsed, err := reference.ParseNormalizedNamed(userSpecifiedImage)
+	if err != nil {
+		return fmt.Errorf("parsing user specified image: %w", err)
+	}
+	userSpecifiedImageParsed = reference.TagNameOnly(userSpecifiedImageParsed)
+
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return fmt.Errorf("create policy context: %w", err)
+	}
+
+	localManifestDigest, err := svc.pullLocalManifestDigest(ctx, systemContext, policyContext, userSpecifiedImageParsed)
+	if err != nil {
+		return fmt.Errorf("pull local manifest digest: %w", err)
+	}
+	if localManifestDigest == "" {
+		return nil
```

Comment:
It’s not OK to silently continue on failures.

This effectively gives a malicious registry a way to bypass signature enforcement by maliciously returning an “unauthorized” response.

Reply:
A Kubernetes e2e test is using a private registry and we get no credentials on container creation. Not sure how we should handle that unsupported use case.

Reply:
See elsewhere for how I expect the local manifest digest to be involved.

Reply:
Removed that code path.